basic-linux-keysniffer/io/keyboardEventReader.cpp

```diff
@@ -0,0 +1,35 @@
+//
+// Created by gnir on 6/7/19.
+//
+#include "keyboardEventReader.h"
+
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <cstdlib>
+
+// Local includes
+#include "consts.h"
+#include "utils.h"
+
+// Kernel includes
+#include "/usr/include/linux/input-event-codes.h"
+
+namespace KeyboardEventReader {
+
+    std::ifstream getKeyboardEventFile() {
+        std::ifstream keyboardEventFile(KeyboardEventReader::fileLocation);
+        SnifferIO::checkFileStream(keyboardEventFile, KeyboardEventReader::fileLocation);
+        return keyboardEventFile;
+    }
+
+    input_event getNextKeyPressEvent(std::ifstream& keyboardEventFile) {
+        input_event newEvent{};
```

Comment:
the {} zero-initializes the struct, which you don't need to do since you're reading into it anyway.
just:
    input_event newEvent;
will default-initialize it and will save you the meaningless zero-init.